core/lib/spree/event.rb

```diff
@@ -108,12 +193,21 @@ def subscriber_registry
 
     private
 
+    def listener_names
+      @listeners_names ||= Set.new
+    end
+
     def normalize_name(name)
-     adapter.normalize_name(name)
+      case name
+      when Symbol
+        name.to_s
+      else
+        name
+      end
     end
 
-    def listener_names
-      @listeners_names ||= Set.new
+    def legacy_adapter(adapter)
```

Comment:
Shouldn't this be `legacy_adapter?` Also, did you consider moving this in the adapter itself? I know they have no Base class to inherits from (that would make it easy to set the default for this method to false) but maybe it's a signal we need it?

Reply:
Sure! I just renamed it `legacy_adapter?`.

As for defining a base class for the adapters, I'd respectfully disagree if the reason is to accommodate the `legacy_adapter?` method. This method is here as a hack until we finish the process and get rid of the ASN adapter, so it's okay to look like a hack. TBH I wouldn't like it to leak into other files. As the idea is to eventually leave us only with the `EventBus` adapter, I wouldn't complicate things at this stage by creating a superclass that would act only as an interface definition.

Reply:
I respectfully accept your disagreement here. 👍 🙂